tests/src/testenhancedassumedstrains.cpp

```diff
@@ -27,15 +27,16 @@ int main(int argc, char** argv) {
   using namespace Dune::Functions::BasisFactory;
   auto firstOrderLagrangePrePower2Basis = power<2>(lagrange<1>(), FlatInterleaved());
   auto firstOrderLagrangePrePower3Basis = power<3>(lagrange<1>(), FlatInterleaved());
-  const auto randomlyDistorted          = CornerDistortionFlag::randomlyDistorted;
-  const auto unDistorted                = CornerDistortionFlag::unDistorted;
-
-  t.subTest(
-      testFEElement<EASElement, 2>(firstOrderLagrangePrePower2Basis, "EAS", randomlyDistorted, checkJacobianFunctor));
-  t.subTest(
-      testFEElement<EASElement, 3>(firstOrderLagrangePrePower3Basis, "EAS", randomlyDistorted, checkJacobianFunctor));
-  t.subTest(
-      testFEElement<EASElement, 2>(firstOrderLagrangePrePower2Basis, "EAS", unDistorted, checkCauchyStressFunctor));
+  constexpr auto randomlyDistorted      = CornerDistortionFlag::randomlyDistorted;
+  constexpr auto unDistorted            = CornerDistortionFlag::unDistorted;
+  constexpr auto cubeGeometry           = ElementGeometryFlag::cube;
+
+  t.subTest(testFEElement<EASElement, 2>(firstOrderLagrangePrePower2Basis, "EAS", randomlyDistorted, cubeGeometry,
```

Comment:
Here you already know the dimension, thus testFEElement does not get the second dimension template but gets just `cube(2)`

Reply:
So you want something like 
```cpp
t.subTest(testFEElement<EASElement>(firstOrderLagrangePrePower2Basis, "EAS", randomlyDistorted, cube(2), ...)
```
So the `dim` is not a template argument anymore?

Reply:
Sounds good

Reply:
Okay i can't get it to work. I need the gridDim as a template argument somewhere because I need it as a template argument when creating the Grid. But i can't extract the gridDim statically from `cube(2)` (i think (?)) nor can i get it from the `PreBasis`as this is more or less just a lambda

Reply:
You may contruct a reference element.
auto refEle= referenceElement(cube(2));
refEle should have the information then

Reply:
I think you have to pass in the `dim` as template argument
```cpp
  template<typename T, int dim>
  auto referenceElement(const Dune::GeometryType& gt, Dune::Dim<dim> = {})
  {
    return ReferenceElements<T,dim>::general(gt);
  }
```

Reply:
Ah ok then everything should be a reference element. then if you need the type you can do refEle.type()

Reply:
https://gitlab.dune-project.org/core/dune-geometry/-/blob/master/dune/geometry/referenceelements.hh?ref_type=heads#L127

https://gitlab.dune-project.org/core/dune-geometry/-/blob/master/dune/geometry/referenceelements.hh?ref_type=heads#L168

Reply:
Okay i have found a solution ... PR is ready to merge from my side (if the tests are sucessfull ;) )